File: plugin/src/main/java/net/onelitefeather/pandorascluster/util/BukkitPlayerUtil.java
package net.onelitefeather.pandorascluster.util;

import net.onelitefeather.pandorascluster.api.enums.Permission;
import net.onelitefeather.pandorascluster.api.util.PlayerUtil;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.UUID;

public class BukkitPlayerUtil implements PlayerUtil {

    @Override
    public @Nullable Player getPlayer(@NotNull UUID uuid) {
        return Bukkit.getPlayer(uuid);
    }

    @Override
    public boolean hasPermission(@NotNull UUID uuid, @NotNull String permission) {
        var player = getPlayer(uuid);
        if (player == null) return false;
        return player.hasPermission(permission);
    }

    @Override
    public boolean hasPermission(@NotNull UUID uuid, @NotNull Permission permission) {
        return hasPermission(uuid, permission.getPermissionNode());
    }
}

File: plugin/src/main/kotlin/net/onelitefeather/pandorascluster/util/BukkitPlayerUtil.kt
package net.onelitefeather.pandorascluster.util

import net.onelitefeather.pandorascluster.api.utils.PlayerUtil
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import java.util.*

class BukkitPlayerUtil : PlayerUtil {

    override fun getPlayer(uuid: UUID): Player? = Bukkit.getPlayer(uuid)

    override fun isOnline(uuid: UUID): Boolean = getPlayer(uuid) != null

    override fun hasPermission(uuid: UUID, permission: String): Boolean {
        val player = getPlayer(uuid) ?: return false
        return player.hasPermission(permission)
    }
}